src/js/server/lib/i18n-node.js

```diff
@@ -0,0 +1,154 @@
+ * @param {String} defaultLocale - The default locale.
+ * @return {Object} - A map of messages for a single locale.
+ *
+ */
+gpii.handlebars.i18n.deriveMessageBundleFromRequest = function (request, messageBundles, defaultLocale) {
+    var header = fluid.get(request, "headers.accept-language");
+    return gpii.handlebars.i18n.deriveMessageBundleFromHeader(header, messageBundles, defaultLocale);
+};
+
+/**
+ *
+ * Load our all message bundles from one or more directories and organise them into a single object.
+ *
+ * @param {Array} messageDirs - An array of full or package-relative paths to directories containing message bundles.
+ * @param {String} defaultLocale - The default locale to use for files that do not provide language or locale data.
+ * @return {Object} - A combined bundle of message bundles, keyed by locales and languages.
+ *
+ */
+gpii.handlebars.i18n.loadMessageBundles = function (messageDirs, defaultLocale) {
+    defaultLocale = defaultLocale || "en_us";
+    var messageBundles = {};
+    var resolvedMessageDirs = gpii.express.hb.resolveAllPaths(messageDirs);
+
+    var filesByLocale   = {};
+    var filesByLanguage = {};
+
+    fluid.each(resolvedMessageDirs, function (messageDir) {
+        var resolvedPath = fluid.module.resolvePath(messageDir);
+        try {
+            var files = fs.readdirSync(resolvedPath);
+            fluid.each(files, function (filename) {
+                var fullPath = path.resolve(resolvedPath, filename);
+                var matchesLocale = filename.toLowerCase().match(/.+[-_]([a-z]{2}[-_][a-z]{2})\.json5?/);
+                var matchesLanguage = filename.toLowerCase().match(/.+[-_]([a-z]{2})\.json5?/);
+
+                if (matchesLocale) {
+                    var preferredLocale = matchesLocale[1];
+                    if (!filesByLocale[preferredLocale]) { filesByLocale[preferredLocale] = []; }
+                    filesByLocale[preferredLocale].push(fullPath);
+                }
+                else if (matchesLanguage) {
+                    var preferredLanguage = matchesLanguage[1];
+                    if (!filesByLanguage[preferredLanguage]) { filesByLanguage[preferredLanguage] = []; }
+                    filesByLanguage[preferredLanguage].push(fullPath);
+                }
+                else {
+                    if (!filesByLocale[defaultLocale]) { filesByLocale[defaultLocale] = []; }
+                    filesByLocale[defaultLocale].push(fullPath);
+                }
+            });
+        }
+        catch (error) {
+            fluid.log("Error loading directory '", resolvedPath, "':", error);
+        }
+    });
+
+    // Load all locales, propagating data back to the language as well.
+    fluid.each(filesByLocale, function (files, locale) {
+        fluid.each(files, function (fullPath) {
+            var data = JSON5.parse(fs.readFileSync(fullPath, "utf8"));
+            messageBundles[locale] = fluid.merge({}, messageBundles[locale], data);
```

Comment:
fluid.extend(true, ...), our punch-through of $.extend, is a better option here. fluid.merge will probably go away in future revisions and for a long time has only really been intended for component options.

Reply:
Done, throughout.